BlackNectar/BlackNectar/StoresTableViewController.swift

```diff
@@ -284,30 +271,95 @@ extension StoresTableViewController {
     
 }
 
-//MARK - UI Screen Pan Gesture Code
+//MARK: Navigation Code
+extension StoresTableViewController {
+    
+    internal func navigateWithDrivingDirections(toStore store: StoresInfo) {
+        
+        let appleMapsLaunchOptions = [MKLaunchOptionsDirectionsModeKey: MKLaunchOptionsDirectionsModeKey]
+        
+        let storePlacemark = MKPlacemark(coordinate: store.location, addressDictionary: ["\(title)" : store.storeName])
+        let storePin = MKMapItem(placemark: storePlacemark)
+        storePin.name = store.storeName
+        
+        storePin.openInMaps(launchOptions: appleMapsLaunchOptions)
+        
+    }
+    
+}
+
+//MARK: Prepare and Perform Segue Code
+extension StoresTableViewController {
+    
+    override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
+        
+        if segue.identifier == "mapViewSegue" {
+            
+            let destination = segue.destination as? StoresMapViewController
+            
+            destination?.distance = distanceFilter
+            destination?.onlyShowOpenStores = self.onlyShowOpenStores
+            destination?.showRestaurants = self.showRestaurants
+            destination?.showStores = self.showStores
+            destination?.storesInMapView = self.stores
+            
+        }
+    }
+    
+    @IBAction func mapButtonTapped(_ sender: Any) {
+        
+        performSegue(withIdentifier: "mapViewSegue", sender: nil)
+        
+    }
+    
+}
+
+//MARK: User Preferences Code
+extension StoresTableViewController {
+    
+    func loadDefaultValues() {
+        
+        onlyShowOpenStores = UserPreferences.instance.isOpenNow
+        showRestaurants = UserPreferences.instance.isRestaurant
+        distanceFilter = UserPreferences.instance.distanceFilter
+        showStores = UserPreferences.instance.isStore
+        
+    }
+    
+}
+
+//MARK: SWReveal Delegate
+extension StoresTableViewController: SWRevealViewControllerDelegate {
+    func revealController(_ revealController: SWRevealViewController!, panGestureRecognizerShouldRecognizeSimultaneouslyWith otherGestureRecognizer: UIGestureRecognizer!) -> Bool {
```

Comment:
I think you can remove this extension and the stuff in it. I was experimenting with it.